services/web/server/src/simcore_service_webserver/security/_authz.py

```diff
@@ -0,0 +1,87 @@
+""" AUTHoriZation (auth) policy:
+"""
+import logging
+
+from aiohttp import web
+from aiohttp_security.abc import AbstractAuthorizationPolicy
+from expiringdict import ExpiringDict
+
+from ..db.plugin import get_database_engine
+from ._authz_access_model import OptionalContext, RoleBasedAccessModel, check_access
+from ._authz_db import UserInfoDict, get_active_user_or_none
+from ._identity import IdentityStr
+
+_logger = logging.getLogger(__name__)
+
+
+class AuthorizationPolicy(AbstractAuthorizationPolicy):
+    def __init__(self, app: web.Application, access_model: RoleBasedAccessModel):
+        self._app = app
+        self._access_model = access_model
+        self._cache = ExpiringDict(max_len=100, max_age_seconds=10)
```

Comment:
question: why you do not use aiocache? I find it so simple and versatile. also it is possible to disable it with an ENV for testing. is just used with a decorator.

Reply:
Good idea. When we built this we did not know (or perhaps did not exist `aiocache`) I will see how can I reuse it

Reply:
Done

Reply:
@sanderegg can you please have a look at the new implemenation with aiocache?